File: backend/app/models/anime.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class AnimeBase(BaseModel):
    anime_id: int
    title: str
    synopsis: Optional[str] = None
    main_picture: Optional[str] = None
    score: Optional[float] = None
    genres: List[str] = []
    studios: List[str] = []
    type: Optional[str] = None
    episodes: Optional[int] = None
    status: Optional[str] = None

class AnimeRecommendation(AnimeBase):
    similarity_score: float

class RecommendationRequest(BaseModel):
    query: Optional[str] = None
    anime_id: Optional[int] = None
    top_n: int = 10
    use_rag: bool = True
    filters: Dict[str, Any] = Field(default_factory=dict)

class RecommendationResponse(BaseModel):
    recommendations: List[AnimeRecommendation]
    metadata: Dict[str, Any] = Field(default_factory=dict)

File: frontend/src/types/anime.ts
export const MEDIA_TYPES = ["anime", "movies", "manga"] as const;
export type MediaType = (typeof MEDIA_TYPES)[number];

export interface Anime {
  anime_id: number;
  title: string;
  synopsis?: string;
  main_picture?: string;
  score?: number;
  genres: string[];
  studios?: string[];
  type?: string;
  episodes?: number;
  status?: string;
  media_type?: string;
  similarity_score?: number;
  metadata?: Record<string, unknown>;
}

export interface RecommendationResponse {
  recommendations: Anime[];
  total?: number;
  metadata?: Record<string, unknown>;
  query_intent?: {
    search_terms: string;
    genres: string[];
    min_score?: number;
    is_recommendation: boolean;
  };
}
